Migration via migration tool failed

At the end of the migration tool run I got:
```
[1/2] 🌀  Converting Package...
Out directory: /home/vaadin/tmp/spring-secured-vaadin/target/migration
Unhandled Rejection at: Promise  Promise {
  <rejected> TypeError: url.replace is not a function
    at Object.replaceHtmlExtensionIfFound (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/urls/util.ts:32:14)
    at Object.generatePackageJson (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/package-manifest.ts:205:24)
    at Object.<anonymous> (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/convert-package.ts:133:23)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/lib/convert-package.js:17:58)
    at <anonymous> }  reason:  TypeError: url.replace is not a function
    at Object.replaceHtmlExtensionIfFound (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/urls/util.ts:32:14)
    at Object.generatePackageJson (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/package-manifest.ts:205:24)
    at Object.<anonymous> (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/src/convert-package.ts:133:23)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vaadin/tmp/spring-secured-vaadin/target/migration/node_modules/polymer-modulizer/lib/convert-package.js:17:58)
    at <anonymous>
[WARNING] Don't know how to resolve Html import 'frontend://bower_components/iron-form/iron-form.html'
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:22 min
[INFO] Finished at: 2019-08-09T13:33:28+02:00
[INFO] Final Memory: 35M/521M
[INFO] ------------------------------------------------------------------------
```

When running the application I get:
```
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '/iron-form/iron-form.js' in '/home/vaadin/tmp/spring-secured-vaadin/target/frontend'
 @ ../target/frontend/generated-flow-imports.js 15:0-33
```

The updated code:
```
@JsModule("/iron-form/iron-form.js")
 public class LoginView extends VerticalLayout {
```

--- a/flow-migration/src/main/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStep.java
+++ b/flow-migration/src/main/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStep.java
@@ -270,18 +270,10 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
             result = result.replaceAll(
                     String.format("\"%s\"", Pattern.quote(path)),
                     String.format("\"%s\"", rewritePath(path,
-                            externalComponent -> getLogger().warn(
-                                    "External bower component {} is imported in the {} file, "
-                                            + "a converted '@JsModule' "
-                                            + "annotation requires also a `@NpmPackage` annotation with "
-                                            + "a module name and a version. The migrated project won't "
-                                            + "be built without this information.",
-                                    externalComponent, javaFile.getPath()),
-                            nonVaadinComponentPath -> getLogger().error(
-                                    "In {} file, added a JS module import '@JsModule(\"{}\")' "
-                                            + "that you need to manually map to the correct package vendor from npm",
-                                    javaFile.getPath(),
-                                    nonVaadinComponentPath))));
+                            externalComponent -> handleBowerComponentImport(
+                                    javaFile, externalComponent),
+                            nonVaadinComponentPath -> handleNonVaadinComponent(
+                                    javaFile, nonVaadinComponentPath))));
         }
         return result;
     }
@@ -295,6 +287,25 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         Pattern pattern = compiledReplacePatterns.computeIfAbsent(patternKey,
                 key -> Pattern.compile(regexp));
         return pattern.matcher(content).replaceAll(replacement);
+    }
+
+    private void handleBowerComponentImport(File javaFile,
+            String bowerComponentPath) {
+        getLogger().warn(
+                "External bower component {} is imported in the {} file, "
+                        + "a converted '@JsModule' "
+                        + "annotation requires also a `@NpmPackage` annotation with "
+                        + "a module name and a version. The migrated project won't "
+                        + "be built without this information.",
+                bowerComponentPath, javaFile.getPath());
+    }
+
+    private void handleNonVaadinComponent(File javaFile,
+            String nonVaadinComponentPath) {
+        getLogger().error(
+                "In {} file, added a JS module import '@JsModule(\"{}\")' "
+                        + "that you need to manually map to the correct package vendor from npm",
+                javaFile.getPath(), nonVaadinComponentPath);
     }
 
     private String rewritePath(String path,

--- a/flow-migration/src/main/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStep.java
+++ b/flow-migration/src/main/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStep.java
@@ -25,6 +25,7 @@ import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.function.Consumer;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
@@ -53,6 +54,9 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
     private static final String CSS_EXTENSION = ".css";
     private final URL compiledClassesURL;
     private final Collection<File> sourceRoots;
+
+    private static final String BOWER_COMPONENT_PREFIX = AbstractCopyResourcesStep.BOWER_COMPONENTS
+            + "/";
 
     private static final String CLASS_DECLARATION_PATTERN = "(\\s|public|final|abstract|private|static|protected)*\\s+class\\s+%s(|<|>|\\?|\\w|\\s|,|\\&)*\\s+((extends\\s+(\\w|<|>|\\?|,)+)|(implements\\s+(|<|>|\\?|\\w|\\s|,)+( ,(\\w|<|>|\\?|\\s|,))*))?\\s*\\{";
 
@@ -235,7 +239,7 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         for (Entry<Class<? extends Annotation>, Collection<String>> entry : annotations
                 .entrySet()) {
 
-            beforeClassDeclaration = rewrite(beforeClassDeclaration,
+            beforeClassDeclaration = rewrite(javaFile, beforeClassDeclaration,
                     entry.getKey(), entry.getValue());
         }
 
@@ -250,7 +254,7 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         }
     }
 
-    private String rewrite(String content,
+    private String rewrite(File javaFile, String content,
             Class<? extends Annotation> annotation, Collection<String> paths) {
         String result = content;
         // replace FQN first
@@ -265,7 +269,19 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         for (String path : paths) {
             result = result.replaceAll(
                     String.format("\"%s\"", Pattern.quote(path)),
-                    String.format("\"%s\"", rewritePath(path)));
+                    String.format("\"%s\"", rewritePath(path,
+                            externalComponent -> getLogger().warn(
+                                    "External bower component {} is imported in the {} file, "
+                                            + "a converted '@JsModule' "
+                                            + "annotation requires also a `@NpmPackage` annotation with "
+                                            + "a module name and a version. The migrated project won't "
+                                            + "be built without this information.",
+                                    externalComponent, javaFile.getPath()),
+                            nonVaadinComponentPath -> getLogger().error(
+                                    "In {} file, added a JS module import '@JsModule(\"{}\")' "
+                                            + "that you need to manually map to the correct package vendor from npm",
+                                    javaFile.getPath(),
+                                    nonVaadinComponentPath))));
         }
         return result;
     }
@@ -281,7 +297,9 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         return pattern.matcher(content).replaceAll(replacement);
     }
 
-    private String rewritePath(String path) {
+    private String rewritePath(String path,
+            Consumer<String> externalComponentHandler,
+            Consumer<String> nonVaadinComponentHandler) {
         String result = path;
         result = rewriteExtension(result, HTML_EXTENSION);
         result = rewriteExtension(result, CSS_EXTENSION);
@@ -293,14 +311,17 @@ public class RewriteLegacyAnnotationsStep extends ClassPathIntrospector {
         result = removePrefix(result,
                 ApplicationConstants.CONTEXT_PROTOCOL_PREFIX);
 
-        if (result.startsWith(AbstractCopyResourcesStep.BOWER_COMPONENTS)) {
+        if (result.startsWith(BOWER_COMPONENT_PREFIX)) {
             result = result.substring(
                     AbstractCopyResourcesStep.BOWER_COMPONENTS.length());
+            externalComponentHandler.accept(result);
             if (result.startsWith("/vaadin")) {
                 result = "@vaadin" + result;
             } else {
+                result = "NPM_VENDOR" + result;
                 getLogger().warn("Don't know how to resolve Html import '{}'",
                         path);
+                nonVaadinComponentHandler.accept(result);
             }
         } else if (result.startsWith("/")) {
             result = "." + result;

--- a/flow-migration/src/test/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStepTest.java
+++ b/flow-migration/src/test/java/com/vaadin/flow/migration/RewriteLegacyAnnotationsStepTest.java
@@ -43,6 +43,7 @@ import com.vaadin.flow.migration.samplecode.Component2;
 import com.vaadin.flow.migration.samplecode.Component3;
 import com.vaadin.flow.migration.samplecode.EnclosingClassWithNestedClass;
 import com.vaadin.flow.migration.samplecode.EnclosingClassWithNestedClass.NestedComponent;
+import com.vaadin.flow.migration.samplecode.NonVaadinComponent;
 import com.vaadin.flow.migration.samplecode.ShouldNotBeRewritten;
 import com.vaadin.flow.migration.samplecode.StyledComponent;
 import com.vaadin.flow.server.frontend.scanner.ClassFinder;
@@ -256,6 +257,24 @@ public class RewriteLegacyAnnotationsStepTest {
         Assert.assertThat(
                 FileUtils.readFileToString(sourceFile, StandardCharsets.UTF_8),
                 CoreMatchers.containsString("@JsModule(\"./baz.js\")"));
+    }
+
+    @Test
+    public void nonVaadinComponent_htmlImportsAreRewrittenUsing_NPM_VENDOR_Prefix()
+            throws IOException {
+        Mockito.when(finder.getAnnotatedClasses(HtmlImport.class))
+                .thenReturn(Collections.singleton(NonVaadinComponent.class));
+        File sourceFile = makeSourceJavaFile(sourceRoot1,
+                NonVaadinComponent.class);
+        step.rewrite();
+
+        String content = FileUtils.readFileToString(sourceFile,
+                StandardCharsets.UTF_8);
+        Assert.assertThat(content, CoreMatchers.allOf(
+                CoreMatchers.containsString(
+                        "import com.vaadin.flow.component.dependency.JsModule;"),
+                CoreMatchers.containsString(
+                        "@JsModule(\"NPM_VENDOR/non-vaadin/non-vaadin.js\")")));
     }
 
     private File makeSourceJavaFile(File root, Class<?> clazz)

--- a/flow-migration/src/test/java/com/vaadin/flow/migration/samplecode/NonVaadinComponent.java
+++ b/flow-migration/src/test/java/com/vaadin/flow/migration/samplecode/NonVaadinComponent.java
@@ -1,0 +1,23 @@
+/*
+ * Copyright 2000-2018 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.migration.samplecode;
+
+import com.vaadin.flow.component.dependency.HtmlImport;
+
+@HtmlImport("bower_components/non-vaadin/non-vaadin.html")
+public class NonVaadinComponent {
+
+}

